### ISSUE TITLE
[docdb] Investigate master side issues with tablet splitting

### ISSUE THREAD

**Issue by bmatican** [yugabyte/yugabyte-db#5390]
All of this is running a hash based Cassandra workload (ie: our sample app for `CassandraBatchTimeseries`)

Two main issues observed so far cc @ttyusupov 
- immediately after split, `list_tablets` no longer shows the parent tablet
- we can actually call split on the same parent tablet and that will make `list_tablets` no longer show the previous children, or the already-split parent tablet...

Flush the table, so we can split
```
17:50 ~/code/yugabyte-db [master] $ ./build/latest/bin/yb-admin --init_master_addrs 127.0.0.1:7100 flush_table ybdemo_keyspace batch_ts_metrics_raw
Flushed [ybdemo_keyspace.batch_ts_metrics_raw] tables.
```
List tablets, observe 6 tablet UUIDs
```
17:51 ~/code/yugabyte-db [master] $ ./build/latest/bin/yb-admin --init_master_addrs 127.0.0.1:7100 list_tablets ybdemo_keyspace batch_ts_metrics_raw
Tablet-UUID                             Range                                                           Leader-IP               Leader-UUID
08476b9e924448749d551645cda86dc2        partition_key_start: "" partition_key_end: "*\252"              127.0.0.2:9100          cf7849a2d645438b98c53632aea1471c
dd9949cd243641b583aff0ef42dfc08e        partition_key_start: "*\252" partition_key_end: "UT"            127.0.0.2:9100          cf7849a2d645438b98c53632aea1471c
ab4bebdf0fc5475bbfccbee519d95fb4        partition_key_start: "UT" partition_key_end: "\177\376"         127.0.0.2:9100          cf7849a2d645438b98c53632aea1471c
3ee7ecfe71b844ac9de16715e8b28583        partition_key_start: "\177\376" partition_key_end: "\252\250"   127.0.0.3:9100          9741d3436b684b649a637044dcaa47d9
963b127d4dcb420da5a9465aa668044f        partition_key_start: "\252\250" partition_key_end: "\325R"      127.0.0.1:9100          dc8371008d2c44b29e6aff850fa05924
051f1250ce0443c1a0931e49c00ae334        partition_key_start: "\325R" partition_key_end: ""              127.0.0.1:9100          dc8371008d2c44b29e6aff850fa05924
```
Pick a tablet UUID and call split for it, eg: `08476b9e924448749d551645cda86dc2`
```
17:51 ~/code/yugabyte-db [master] $ ./build/latest/bin/yb-admin --init_master_addrs 127.0.0.1:7100 split_tablet 08476b9e924448749d551645cda86dc2
Response:
```
^ Note the response is empty(?)

List tablets again, quickly, observe only 5 tablets -- the one we just split is not in the list
```
17:51 ~/code/yugabyte-db [master] $ ./build/latest/bin/yb-admin --init_master_addrs 127.0.0.1:7100 list_tablets ybdemo_keyspace batch_ts_metrics_raw
Tablet-UUID                             Range                                                           Leader-IP               Leader-UUID
dd9949cd243641b583aff0ef42dfc08e        partition_key_start: "*\252" partition_key_end: "UT"            127.0.0.2:9100          cf7849a2d645438b98c53632aea1471c
ab4bebdf0fc5475bbfccbee519d95fb4        partition_key_start: "UT" partition_key_end: "\177\376"         127.0.0.2:9100          cf7849a2d645438b98c53632aea1471c
3ee7ecfe71b844ac9de16715e8b28583        partition_key_start: "\177\376" partition_key_end: "\252\250"   127.0.0.3:9100          9741d3436b684b649a637044dcaa47d9
963b127d4dcb420da5a9465aa668044f        partition_key_start: "\252\250" partition_key_end: "\325R"      127.0.0.1:9100          dc8371008d2c44b29e6aff850fa05924
051f1250ce0443c1a0931e49c00ae334        partition_key_start: "\325R" partition_key_end: ""              127.0.0.1:9100          dc8371008d2c44b29e6aff850fa05924
```

List tablets again a bit later, confirm we see 7 tablets now -- no parent, but the 2 children
```
17:51 ~/code/yugabyte-db [master] $ ./build/latest/bin/yb-admin --init_master_addrs 127.0.0.1:7100 list_tablets ybdemo_keyspace batch_ts_metrics_raw
Tablet-UUID                             Range                                                           Leader-IP               Leader-UUID
03d34d9b1e25473aa157939d88465149        partition_key_start: "" partition_key_end: "\025U"              127.0.0.3:9100          9741d3436b684b649a637044dcaa47d9
5833746b903049edb0c919f69b21f5cb        partition_key_start: "\025U" partition_key_end: "*\252"         127.0.0.2:9100          cf7849a2d645438b98c53632aea1471c
dd9949cd243641b583aff0ef42dfc08e        partition_key_start: "*\252" partition_key_end: "UT"            127.0.0.2:9100          cf7849a2d645438b98c53632aea1471c
ab4bebdf0fc5475bbfccbee519d95fb4        partition_key_start: "UT" partition_key_end: "\177\376"         127.0.0.2:9100          cf7849a2d645438b98c53632aea1471c
3ee7ecfe71b844ac9de16715e8b28583        partition_key_start: "\177\376" partition_key_end: "\252\250"   127.0.0.3:9100          9741d3436b684b649a637044dcaa47d9
963b127d4dcb420da5a9465aa668044f        partition_key_start: "\252\250" partition_key_end: "\325R"      127.0.0.1:9100          dc8371008d2c44b29e6aff850fa05924
051f1250ce0443c1a0931e49c00ae334        partition_key_start: "\325R" partition_key_end: ""              127.0.0.1:9100          dc8371008d2c44b29e6aff850fa05924
```

Split same parent tablet again, to see if it works, confirm it does, but something breaks
```
17:51 ~/code/yugabyte-db [master] $ ./build/latest/bin/yb-admin --init_master_addrs 127.0.0.1:7100 split_tablet 08476b9e924448749d551645cda86dc2
Response:
17:51 ~/code/yugabyte-db [master] $ ./build/latest/bin/yb-admin --init_master_addrs 127.0.0.1:7100 list_tablets ybdemo_keyspace batch_ts_metrics_raw
Tablet-UUID                             Range                                                           Leader-IP               Leader-UUID
dd9949cd243641b583aff0ef42dfc08e        partition_key_start: "*\252" partition_key_end: "UT"            127.0.0.2:9100          cf7849a2d645438b98c53632aea1471c
ab4bebdf0fc5475bbfccbee519d95fb4        partition_key_start: "UT" partition_key_end: "\177\376"         127.0.0.2:9100          cf7849a2d645438b98c53632aea1471c
3ee7ecfe71b844ac9de16715e8b28583        partition_key_start: "\177\376" partition_key_end: "\252\250"   127.0.0.3:9100          9741d3436b684b649a637044dcaa47d9
963b127d4dcb420da5a9465aa668044f        partition_key_start: "\252\250" partition_key_end: "\325R"      127.0.0.1:9100          dc8371008d2c44b29e6aff850fa05924
051f1250ce0443c1a0931e49c00ae334        partition_key_start: "\325R" partition_key_end: ""              127.0.0.1:9100          dc8371008d2c44b29e6aff850fa05924
```

**Comment by bmatican** [comment]
Another note: the master can end up sending `CreateTablet` calls for the split child tablets, due to them being in CREATING state.

**Comment by bmatican** [comment]
This was basically fixed by 53c383f014bb426498368ff9ef1546b445fa23ba, making sure that tablets in CREATING don't get issued a `CreateTablet` call
